File: frontend/src/types/etl.ts
export interface PipelineRun {
  id: number;
  pipeline_id: string;
  status: string;
  started_at: string;
  finished_at?: string | null;
  row_count?: number | null;
  error_message?: string | null;
}

export interface PipelineRunCreate {
  pipeline_id: string;
}

File: backend/app/schemas/etl.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class PipelineRunCreate(BaseModel):
    pipeline_id: str


class PipelineRunResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    pipeline_id: str
    status: str
    started_at: datetime
    finished_at: datetime | None = None
    row_count: int | None = None
    error_message: str | None = None
